File: 2023/langs/go/1/solution1.go
package main

import (
	"log"
	"slices"
	"strconv"
	"strings"
)

func solution1(calibrationData []string) int {
	numbers := strings.Split("1234567890", "")

	total := 0

	for _, value := range calibrationData {
		digits := ""
		for i := 0; i < len(value); i++ {
			digit := string(value[i])
			if !slices.Contains(numbers, digit) {
				continue
			}

			digits += digit
			break
		}

		for i := len(value) - 1; i >= 0; i-- {
			digit := string(value[i])
			if !slices.Contains(numbers, digit) {
				continue
			}

			digits += digit
			break
		}

		i, err := strconv.Atoi(digits)
		if err != nil {
			log.Fatalln("Couldn't parse digits to integer")
		}

		total += i
	}

	return total
}

File: 2023/langs/kotlin/src/main/kotlin/Solution1.kt
fun solution1(): Int {
    val numbers = "1234567890"

    var total = 0

    calibrationData.forEach {
        var digits = ""
        for (i in 0..it.length) {
            val digit = it[i]
            if (!numbers.contains(digit)) continue

            digits += digit
            break
        }

        for (i in it.indices.reversed()) {
            val digit = it[i]
            if (!numbers.contains(digit)) continue

            digits += digit
            break
        }

        total += digits.toInt()
    }

    return total
}